src/model.cpp

```diff
@@ -39,12 +40,27 @@ using LibMatrix::vec2;
 using LibMatrix::vec3;
 using LibMatrix::uvec3;
 
-#define read_or_fail(file, dst, size) do { \
-    file.read(reinterpret_cast<char *>((dst)), (size)); \
-    if (file.gcount() < (std::streamsize)(size)) { \
+static inline void read_from_file(std::istream& file, void *dst, size_t nmemb, size_t size) {
+  file.read(reinterpret_cast<char *>(dst), nmemb * size);
```

Comment:
Indent size is 4 (not 2). Similarly below.

Reply:
Done